src/gtc/python/npir_gen.py

```diff
@@ -230,18 +230,21 @@ def visit_EmptyTemp(
 
     VectorTemp = FormatTemplate("{name}_")
 
-    def visit_MaskBlock(self, node: npir.MaskBlock, **kwargs) -> Union[str, Collection[str]]:
+    def visit_MaskBlock(
+        self, node: npir.MaskBlock, *, is_serial: bool = False, **kwargs
+    ) -> Union[str, Collection[str]]:
         if isinstance(node.mask, npir.FieldSlice):
             mask_def = ""
         elif isinstance(node.mask, npir.BroadCast):
             mask_name = node.mask_name
             mask = self.visit(node.mask)
-            mask_def = f"{mask_name}_ = np.full((I - i, J - j, K - k), {mask})\n"
+            ksize = "1" if is_serial else "K - k"
+            mask_def = f"{mask_name}_ = np.full((I - i, J - j, {ksize}), {mask})\n"
```

Comment:
```suggestion
            k_size = "1" if is_serial else "K - k"
            mask_def = f"{mask_name}_ = np.full((I - i, J - j, {k_size}), {mask})\n"
```

just for consistency / searchability.